pkg/filters/proxies/loadbalance.go

```diff
@@ -108,6 +111,8 @@ func (glb *GeneralLoadBalancer) Init(
 			lbp = &IPHashLoadBalancePolicy{}
 		case LoadBalancePolicyHeaderHash:
 			lbp = &HeaderHashLoadBalancePolicy{spec: glb.spec}
+		case LoadBalancePolicyCookieHash:
+			lbp = &HeaderHashLoadBalancePolicy{spec: &LoadBalanceSpec{HeaderHashKey: "Set-Cookie"}}
```

Comment:
`Set-Cookie` is in response header that server send to client. In this place, client sends requests to server. I think maybe `Cookie` header is more appropriate. 

As show in https://docs.oracle.com/en- us/iaas/Content/Balance/Reference/sessionpersistence.htm#cook__HowItWorks

```
The Load Balancer service calculates a hash of the configured cookie and other request parameters, 
and sends that value to the client in a cookie. 
The value stored in the cookie enables the service to route subsequent client requests to the correct backend server. 
If your backend servers change any of the defined cookies, 
the service recomputes the cookie's value and resends it to the client.
```

My understand for this is that load balancer use `Set-Cookie` header to set some information to client. And in later requests, get this information from `Cookie` header and use this information to route to correct server? 

And for cookie hash, do we support to route based on a specific cookie or route based on all cookies? When use `header.Get()` is always return first cookie...

Reply:
Updated